ReSwift/CoreTypes/Store.swift

```diff
@@ -17,14 +17,26 @@ import Foundation
  */
 public class Store<State: StateType>: StoreType {
 
+    typealias SubscriptionType = Subscription<State>
+
     // TODO: Setter should not be public; need way for store enhancers to modify appState anyway
 
     /*private (set)*/ public var state: State! {
         didSet {
-            subscriptions.forEach {
+            var indexesToRemove: [Int] = []
+            for (index, subscription) in subscriptions.enumerate() {
+                guard let subscriber = subscription.subscriber else {
+                    indexesToRemove.append(index)
+                    return
```

Comment:
The escape route here should be a `continue` and not a `return`. With the current code subscribers after a no-longer-existent subscriber would miss a state update.

I think it would be best to separate these two pieces of logic:
1) Filter out subscriptions that have lost their reference to a subscriber
2) Notify all remaining subscribers

Could look something like this:

```
subscriptions = subscriptions.filter { $0.subscriber != nil }
subscriptions.forEach {
     $0.subscriber!._newState(subscription.selector?(state) ?? state)
}
```
Since the use of the implicit unwrapped optional here is very close to the line that filters out nil-values, I think it is fine to use and shouldn't through off folks reading the code.